plugins/common/http/config.go

```diff
@@ -0,0 +1,73 @@
+package httpconfig
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/common/proxy"
+	"github.com/influxdata/telegraf/plugins/common/tls"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/clientcredentials"
+)
+
+// Common HTTP client struct.
+type HTTPClientConfig struct {
+	// OAuth2 Credentials
+	ClientID     string   `toml:"client_id"`
+	ClientSecret string   `toml:"client_secret"`
+	TokenURL     string   `toml:"token_url"`
```

Comment:
Can we please also seperate this into something like `plugins/common/oauth/oauth.go`. It might be also useful for other transports or in existing plugins.

Reply:
Yup.